Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/SharedOpaqueDirectoryTests.cs

```diff
@@ -59,8 +60,14 @@ public void SharedOpaqueDirectoryConsumptionCachingBehavior()
             var pipB = SchedulePipBuilder(builderB);
 
             // B should be able to consume the file in the opaque directory. Second build should have both cached
-            RunScheduler().AssertCacheMiss(pipA.Process.PipId, pipB.Process.PipId);
+            var result = RunScheduler().AssertCacheMiss(pipA.Process.PipId, pipB.Process.PipId);
+            AssertWritesJournaled(result, pipA, outputInSharedOpaque);

```

Comment:
For now I only added this single integration test.

I'll try to hijack some low-level test function and there uniformly check that all outputs into shared opaque directories are properly journaled.

Reply:
I would put this in its own integration test. Now it is mixed up with a test that checks caching behavior

---
In reply to: [333684941](https://github.com/microsoft/BuildXL/pull/1029#discussion_r333684941) [](ancestors = 333684941)

Reply:
agreed; this was a temp solution just to have at least one check in an integration test